message_center_compassion/model/child_properties.py

```diff
@@ -8,21 +8,17 @@
 #    The licence is in the file __openerp__.py
 #
 ##############################################################################
-from openerp.osv import orm
+from openerp import models
 
 
-class child_properties(orm.Model):
+class child_properties(models.Model):
     """ Write contracts in Biennial State when a picture is attached to a
         Case Study. """
     _inherit = 'compassion.child.property'
 
-    def attach_pictures(self, cr, uid, ids, pictures_id, context=None):
-        res = super(child_properties, self).attach_pictures(
-            cr, uid, ids, pictures_id, context)
-        if not isinstance(ids, list):
-            ids = [ids]
-        property = self.browse(cr, uid, ids, context)[0]
-        for contract in property.child_id.sponsorship_ids:
+    def attach_pictures(self, pictures_id):
+        res = super(child_properties, self).attach_pictures(pictures_id)
+        for contract in self.child_id.sponsorship_ids:

```

Comment:
May be prettier to filter `sponsorship_ids` by contract state `('waiting', 'active', 'mandate')` and then call `new_biennal()` on the returned RecordSet